activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -145,6 +145,10 @@ def self.define_validations(model, reflection)
         reflection.options[:optional] = !reflection.options.delete(:required)
       end
 
+      if reflection.options.key?(:message)
+        validation_message = reflection.options.delete(:message)
```

Comment:
What do you think about naming this key `:validation_message` or something similar? I believe this will improve the readability and really reflect what the usage of this value will be.

```ruby
belongs_to :user, required: true, validation_message: "should be set"
```